components/i2c/include/i2c.hpp

```diff
@@ -87,23 +87,26 @@ class I2c : public espp::BaseComponent {
     // Make copy of variables for easy capture
     i2c_port_t i2c_port = config_.port;
     int core_id = config_.isr_core_id;
-    if (core_id < 0) {
+    if (core_id <= 0) {
```

Comment:
You don't know that this function will be called on core 0 without checking so I wouldn't do the direct driver install in this case.